<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1100</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1100</width>
    <height>800</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout" stretch="4,20">
    <item>
     <layout class="QGridLayout" name="gridLayout_6" rowstretch="0,0">
      <property name="topMargin">
       <number>0</number>
      </property>
      <item row="0" column="0" rowspan="2">
       <widget class="QGroupBox" name="groupBox_4">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>0</height>
         </size>
        </property>
        <property name="title">
         <string>GroupBox</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_7">
         <property name="topMargin">
          <number>0</number>
         </property>
         <item row="1" column="2" rowspan="3">
          <widget class="QPushButton" name="pushButton_15">
           <property name="text">
            <string>ADD 500 ENCODERS</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QLabel" name="label_60">
           <property name="text">
            <string>2.34456.67.6</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QPushButton" name="pushButton_11">
           <property name="text">
            <string>Read Config from Device</string>
           </property>
          </widget>
         </item>
         <item row="1" column="3" rowspan="3">
          <widget class="QPushButton" name="pushButton_TEST_MAIN_BUTTON">
           <property name="text">
            <string>READ CONFIG</string>
           </property>
          </widget>
         </item>
         <item row="0" column="2">
          <widget class="QPushButton" name="pushButton_16">
           <property name="text">
            <string>Write Config to Device</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1" rowspan="3">
          <widget class="QPushButton" name="pushButton_13">
           <property name="text">
            <string>Load Config from File</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QComboBox" name="comboBox_2"/>
         </item>
         <item row="0" column="3">
          <widget class="QPushButton" name="pushButton_TEST_2_BUTTON">
           <property name="text">
            <string>WRITE CONFIG</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QLabel" name="label_59">
           <property name="text">
            <string>F103Joy</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="baseSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="font">
       <font>
        <pointsize>10</pointsize>
       </font>
      </property>
      <property name="styleSheet">
       <string notr="true">QTabWidget::pane
{
border-top: 1px solid rgb(190, 190, 190);
border-left: 1px solid rgb(190, 190, 190);
border-right: 1px solid rgb(190, 190, 190);
border-bottom: 1px solid rgb(190, 190, 190);
}</string>
      </property>
      <property name="tabShape">
       <enum>QTabWidget::Rounded</enum>
      </property>
      <property name="currentIndex">
       <number>3</number>
      </property>
      <widget class="QWidget" name="tab_PinConfig">
       <attribute name="title">
        <string> Pin Config </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabPinConfig">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_ButtonConfig">
       <attribute name="title">
        <string> Button Config </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabButtonConfig">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_AxesConfig">
       <attribute name="title">
        <string> Axes Config </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabAxesConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_AxesCurves">
       <attribute name="title">
        <string> Axes Curves </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabAxesCurvesConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_AxesToButtons">
       <attribute name="title">
        <string> Axes to Buttons </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabAxesToButtons">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_ShiftRegisters">
       <attribute name="title">
        <string> Shift Registers </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabShiftRegistersConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_Encoders">
       <attribute name="title">
        <string>Encoders</string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabEncodersConfig">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_LED">
       <attribute name="title">
        <string> LED </string>
       </attribute>
       <layout class="QVBoxLayout" name="layoutV_tabLedConfig">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_AdvancedSettings">
       <attribute name="title">
        <string> Advanced Settings </string>
       </attribute>
       <layout class="QGridLayout" name="gridLayout_11">
        <item row="0" column="0">
         <widget class="AdvancedSettings" name="widget_2" native="true"/>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_Debug">
       <attribute name="title">
        <string> Debug </string>
       </attribute>
       <widget class="QPushButton" name="pushButton_4">
        <property name="geometry">
         <rect>
          <x>30</x>
          <y>10</y>
          <width>75</width>
          <height>23</height>
         </rect>
        </property>
        <property name="text">
         <string>PushButton</string>
        </property>
       </widget>
       <widget class="QScrollArea" name="scrollAreaPhysButton">
        <property name="geometry">
         <rect>
          <x>10</x>
          <y>70</y>
          <width>371</width>
          <height>401</height>
         </rect>
        </property>
        <property name="frameShape">
         <enum>QFrame::StyledPanel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Plain</enum>
        </property>
        <property name="lineWidth">
         <number>1</number>
        </property>
        <property name="horizontalScrollBarPolicy">
         <enum>Qt::ScrollBarAlwaysOff</enum>
        </property>
        <property name="widgetResizable">
         <bool>true</bool>
        </property>
        <widget class="QWidget" name="scrollAreaWidgetContents">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>369</width>
           <height>399</height>
          </rect>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <layout class="QGridLayout" name="gridLayout"/>
          </item>
         </layout>
        </widget>
       </widget>
       <widget class="QLabel" name="label_DeviceName">
        <property name="geometry">
         <rect>
          <x>190</x>
          <y>20</y>
          <width>101</width>
          <height>21</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>12</pointsize>
         </font>
        </property>
        <property name="text">
         <string>TextLabel</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QPushButton" name="button_SendConfig">
        <property name="geometry">
         <rect>
          <x>580</x>
          <y>160</y>
          <width>121</width>
          <height>51</height>
         </rect>
        </property>
        <property name="text">
         <string>SEND CONFIG</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_ID">
        <property name="geometry">
         <rect>
          <x>600</x>
          <y>10</y>
          <width>91</width>
          <height>41</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>16</pointsize>
         </font>
        </property>
        <property name="text">
         <string>hz</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QPushButton" name="button_Prost">
        <property name="geometry">
         <rect>
          <x>390</x>
          <y>450</y>
          <width>16</width>
          <height>16</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 0, 0);</string>
        </property>
        <property name="text">
         <string>X</string>
        </property>
       </widget>
       <widget class="QLabel" name="label_3">
        <property name="geometry">
         <rect>
          <x>400</x>
          <y>0</y>
          <width>121</width>
          <height>20</height>
         </rect>
        </property>
        <property name="text">
         <string>REPORT_ID_JOY count</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QPushButton" name="button_GetConfig">
        <property name="geometry">
         <rect>
          <x>420</x>
          <y>160</y>
          <width>121</width>
          <height>51</height>
         </rect>
        </property>
        <property name="text">
         <string>GET CONFIG</string>
        </property>
       </widget>
       <widget class="QLineEdit" name="lineEdit">
        <property name="geometry">
         <rect>
          <x>410</x>
          <y>20</y>
          <width>101</width>
          <height>31</height>
         </rect>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <property name="readOnly">
         <bool>false</bool>
        </property>
       </widget>
       <widget class="QGroupBox" name="groupBox">
        <property name="geometry">
         <rect>
          <x>470</x>
          <y>260</y>
          <width>181</width>
          <height>151</height>
         </rect>
        </property>
        <property name="title">
         <string>GroupBox</string>
        </property>
        <widget class="QSpinBox" name="spinBoxButtonNumber">
         <property name="geometry">
          <rect>
           <x>65</x>
           <y>50</y>
           <width>51</width>
           <height>20</height>
          </rect>
         </property>
         <property name="maximum">
          <number>999</number>
         </property>
         <property name="value">
          <number>12</number>
         </property>
        </widget>
        <widget class="QLabel" name="label">
         <property name="geometry">
          <rect>
           <x>55</x>
           <y>20</y>
           <width>71</width>
           <height>16</height>
          </rect>
         </property>
         <property name="text">
          <string>Button number</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>90</y>
           <width>75</width>
           <height>23</height>
          </rect>
         </property>
         <property name="text">
          <string>Increase</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_2">
         <property name="geometry">
          <rect>
           <x>50</x>
           <y>120</y>
           <width>75</width>
           <height>23</height>
          </rect>
         </property>
         <property name="text">
          <string>Remove all</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_3">
         <property name="geometry">
          <rect>
           <x>90</x>
           <y>90</y>
           <width>75</width>
           <height>23</height>
          </rect>
         </property>
         <property name="text">
          <string>Decrease</string>
         </property>
        </widget>
       </widget>
       <widget class="QLabel" name="label_DeviceStatus">
        <property name="geometry">
         <rect>
          <x>480</x>
          <y>70</y>
          <width>161</width>
          <height>51</height>
         </rect>
        </property>
        <property name="font">
         <font>
          <pointsize>20</pointsize>
         </font>
        </property>
        <property name="text">
         <string>Connected</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="label_2">
        <property name="geometry">
         <rect>
          <x>205</x>
          <y>0</y>
          <width>71</width>
          <height>16</height>
         </rect>
        </property>
        <property name="text">
         <string>Device name:</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="label_6">
        <property name="geometry">
         <rect>
          <x>500</x>
          <y>460</y>
          <width>20</width>
          <height>20</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true"> border-radius: 10px;
 min-height: 20px;
 min-width: 20px;
background-color: rgb(255, 0, 0);</string>
        </property>
        <property name="text">
         <string>1</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="label_7">
        <property name="geometry">
         <rect>
          <x>530</x>
          <y>460</y>
          <width>20</width>
          <height>20</height>
         </rect>
        </property>
        <property name="styleSheet">
         <string notr="true">  border-radius: 10px;
  min-height: 20px;
  min-width: 20px;
background-color: rgb(13, 204, 0);</string>
        </property>
        <property name="text">
         <string>1</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
       <widget class="QLabel" name="label_VID">
        <property name="geometry">
         <rect>
          <x>310</x>
          <y>30</y>
          <width>47</width>
          <height>13</height>
         </rect>
        </property>
        <property name="text">
         <string>TextLabel</string>
        </property>
       </widget>
       <widget class="QGroupBox" name="groupBox_2">
        <property name="geometry">
         <rect>
          <x>830</x>
          <y>170</y>
          <width>141</width>
          <height>141</height>
         </rect>
        </property>
        <property name="title">
         <string>Languages</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QPushButton" name="button_EngLang">
           <property name="text">
            <string>English</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="button_RusLang">
           <property name="text">
            <string>Русский</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QGroupBox" name="groupBox_12">
        <property name="geometry">
         <rect>
          <x>800</x>
          <y>340</y>
          <width>231</width>
          <height>231</height>
         </rect>
        </property>
        <property name="title">
         <string>Styles</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
        <layout class="QGridLayout" name="gridLayout_10">
         <item row="0" column="1">
          <widget class="QPushButton" name="pushButton_QssQDarkStyle">
           <property name="text">
            <string>qDarkStyle</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="pushButton_QssConsoleStyle">
           <property name="text">
            <string>ConsoleStyle</string>
           </property>
          </widget>
         </item>
         <item row="4" column="1">
          <widget class="QPushButton" name="pushButton_QssMaterialDark">
           <property name="text">
            <string>MaterialDark</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0">
          <widget class="QPushButton" name="pushButton_QssDefault">
           <property name="text">
            <string>Default</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0">
          <widget class="QPushButton" name="pushButton_QssUbuntu">
           <property name="text">
            <string>Ubuntu</string>
           </property>
          </widget>
         </item>
         <item row="3" column="1">
          <widget class="QPushButton" name="pushButton_QssManjaroMix">
           <property name="text">
            <string>ManjaroMix</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QPushButton" name="pushButton_QssAMOLED">
           <property name="text">
            <string>AMOLED</string>
           </property>
          </widget>
         </item>
         <item row="4" column="0">
          <widget class="QPushButton" name="pushButton_QssElegantDark">
           <property name="text">
            <string>ElegantDark</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0">
          <widget class="QPushButton" name="pushButton_QssAqua">
           <property name="text">
            <string>Aqua</string>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="pushButton_QssChinaWhite">
           <property name="text">
            <string>ChinaWhite</string>
           </property>
          </widget>
         </item>
         <item row="5" column="0">
          <widget class="QPushButton" name="pushButton_QssBreezeWhite">
           <property name="text">
            <string>Breeze white</string>
           </property>
          </widget>
         </item>
         <item row="5" column="1">
          <widget class="QPushButton" name="pushButton_QssBreezeDark">
           <property name="text">
            <string>Breeze dark</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>AdvancedSettings</class>
   <extends>QWidget</extends>
   <header>widgets/advancedsettings.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
